### PR TITLE
Increase verbosity when using update syntax with private fields

### LINKED ISSUE #70323: Regression on privacy field of struct used in `..some_struct` syntax
In a recent PR #69942 I made a change to unify the wording of multiple privacy errors, but [accidentally regressed the following case](https://github.com/rust-lang/rust/pull/69942/files#diff-545c4dd451e64756495b0adf2f3f23aa):

```
error[E0451]: field `secret_uid` of struct `foo::S` is private
  --> $DIR/functional-struct-update-respects-privacy.rs:28:49
   |
LL |     let s_2 = foo::S { b: format!("ess two"), ..s_1 }; // FRU ...
   |                                                 ^^^ private field
```
used to be
```
error[E0451]: field `secret_uid` of struct `foo::S` is private
  --> $DIR/functional-struct-update-respects-privacy.rs:28:49
   |
LL |     let s_2 = foo::S { b: format!("ess two"), ..s_1 }; // FRU ...
   |                                                 ^^^ field `secret_uid` is private
```

This one change needs to be reversed while keeping the rest.

### PR DESCRIPTION
**Pull request by jeremystucki**
Resolves #70323

### CODE REVIEW

**Comment by estebank** [comment]
Can you run `./x.py fmt` and squash? There's a missing comma that CI will reject:

```
Diff in /checkout/src/librustc_privacy/lib.rs at line 1023:
         span: Span,            // span of the field pattern, e.g., `x: 0`
         def: &'tcx ty::AdtDef, // definition of the struct or enum
         field: &'tcx ty::FieldDef,
-        in_update_syntax: bool
+        in_update_syntax: bool,
     ) {
```

Otherwise, LGTM.

File: src/librustc_privacy/lib.rs [hunk 1]
```diff
@@ -1038,7 +1045,7 @@ impl<'a, 'tcx> NamePrivacyVisitor<'a, 'tcx> {
                 def.variant_descr(),
                 self.tcx.def_path_str(def.did)
             )
-            .span_label(span, "private field")
+            .span_label(span, label)
```
Comment (estebank) [inline src/librustc_privacy/lib.rs]: I thought `span_label` only took `&str`.

**Comment by jeremystucki** [comment]
> Can you run `./x.py fmt` and squash? There's a missing comma that CI will reject:

Sorry about that

**Pull request merged**